src/java.base/share/classes/java/lang/StackStreamFactory.java

```diff
@@ -539,15 +539,14 @@ protected void initFrameBuffer() {
         @Override
         protected int batchSize(int lastBatchFrameCount) {
             if (lastBatchFrameCount == 0) {
-                // First batch, use estimateDepth if not exceed the large batch size
-                int initialBatchSize = Math.max(walker.estimateDepth()+RESERVED_ELEMENTS, MIN_BATCH_SIZE);
-                return Math.min(initialBatchSize, LARGE_BATCH_SIZE);
+                return walker.estimateDepth() == 0
+                        ? SMALL_BATCH
+                        : Math.min(walker.estimateDepth() + RESERVED_ELEMENTS, LARGE_BATCH_SIZE);
```

Comment:
Without the
`Math.max(walker.estimateDepth()+RESERVED_ELEMENTS, MIN_BATCH_SIZE)`
for estimateDepth = 1, I believe this will now return 2, where previously it returned 3.
Is that OK?

Reply:
yes

Reply:
Sounds good, thanks